### PR TITLE
add Sophora Image AI

### PR DESCRIPTION
**Pull request by blizzy78**
This PR adds a new chart for Sophora Image AI.

### CODE REVIEW

File: charts/sophora-image-ai/values.yaml [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+
+image:
+  repository: docker.subshell.com/sophora/imageai
+  pullPolicy: IfNotPresent
+  # Overrides the image tag whose default is the chart appVersion.
+  tag: "4.0.0"
+
+imagePullSecrets: []
+
+javaOptions: -XX:+ExitOnOutOfMemoryError -XX:+PerfDisableSharedMem
```
Comment (tom-schoener) [inline charts/sophora-image-ai/values.yaml:12]: Whenever the value contains a `:` I'd surround the value with quotation marks.

File: charts/sophora-image-ai/values.yaml [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+
+    spring:
+      cloud:
+        gcp:
+          # GCP project ID
+          project-id: my-project
+
+          credentials:
+            # path to GCP credentials file - do not change
+            location: file:/gcp-credentials/credentials.json
```
Comment (tom-schoener) [inline charts/sophora-image-ai/values.yaml:39]: same here (quotation marks)

File: charts/sophora-image-ai/templates/deployment.yaml [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+  name: {{ include "sophora-image-ai.fullname" . }}
+  labels:
+    {{- include "sophora-image-ai.labels" . | nindent 4 }}
+spec:
+  selector:
+    matchLabels:
+      {{- include "sophora-image-ai.selectorLabels" . | nindent 6 }}
+  template:
+    metadata:
+      {{- with .Values.podAnnotations }}
```
Comment (tom-schoener) [inline charts/sophora-image-ai/templates/deployment.yaml:13]: Whenever the configmap changes the deployment should restart. For this to happen you need to register the config as a "dependency":

```
      annotations:
        checksum/config: {{ include (print $.Template.BasePath "/configmap.yaml") . | sha256sum }}
        {{- with .Values.podAnnotations }}
        {{- toYaml . | nindent 8 }}
        {{- end }}
```

**Review by tom-schoener — CHANGES_REQUESTED** [review]

File: charts/sophora-image-ai/templates/deployment.yaml (hunk 3 above)
Comment (blizzy78) [reply charts/sophora-image-ai/templates/deployment.yaml]: Interesting, I've added it.

File: charts/sophora-image-ai/values.yaml (hunk 1 above)
Comment (blizzy78) [reply charts/sophora-image-ai/values.yaml:12]: I usually don't. To a compliant YAML parser, this should be a regular string.

**Review by tom-schoener — APPROVED** [review]

**Pull request merged**